### PR TITLE
enable sending a token or a SAC payment to a C address

### PR DESCRIPTION
**Pull request by piyalbasu**
#1414

Allowing user to send a payment to a C address using either a token or a SAC

### CODE REVIEW

File: extension/src/popup/components/manageAssets/ChooseAsset/index.tsx [hunk 1]
```diff
@@ -46,7 +46,7 @@ export const ChooseAsset = ({ balances }: ChooseAssetProps) => {
   const isSwap = useIsSwap();
   const isSoroswapEnabled = useIsSoroswapEnabled();
 
-  const managingAssets = assetSelect.type === AssetSelectType.MANAGE;
+  const isManagingAssets = assetSelect.type === AssetSelectType.MANAGE;
```
Comment (piyalbasu) [inline extension/src/popup/components/manageAssets/ChooseAsset/index.tsx:49]: This is just a var name issue that was bugging me , so fixing it here

File: extension/src/popup/components/sendPayment/SendSettings/Settings/index.tsx [hunk 2]
```diff
@@ -85,6 +88,24 @@ export const Settings = ({
   // dont show memo for regular sends to Muxed, or for swaps
   const showMemo = !isSwap && !isMuxedAccount(destination);
   const showSlippage = (isPathPayment || isSwap) && !isSoroswap;
+  const isSendSacToContract =
+    isContractId(destination) &&
+    !isContractId(getAssetFromCanonical(asset).issuer);
```
Comment (piyalbasu) [inline extension/src/popup/components/sendPayment/SendSettings/Settings/index.tsx:93]: The logic here is basically:
If you're sending to a C address, but you're trying to send a classic asset, grab the C address of the SAC

File: extension/src/popup/components/sendPayment/SendSettings/Settings/index.tsx [hunk 3]
```diff
@@ -109,8 +130,10 @@ export const Settings = ({
       return;
     }
 
-    if (isToken) {
-      const assetAddress = asset.split(":")[1];
+    if (isToken || isSendSacToContract) {
```
Comment (piyalbasu) [inline extension/src/popup/components/sendPayment/SendSettings/Settings/index.tsx:133]: Sending a token to a C address generally worked "out of the box" once I allowed a user to enter a C address as a destination 🎉

File: extension/src/popup/components/sendPayment/SendTo/index.tsx [hunk 4]
```diff
@@ -106,7 +107,7 @@ export const SendTo = ({ previous }: { previous: ROUTES }) => {
   );
 
   const [recentAddresses, setRecentAddresses] = useState<string[]>([]);
-  const [validatedPubKey, setValidatedPubKey] = useState("");
+  const [validatedAddress, setValidatedAddress] = useState("");
```
Comment (piyalbasu) [inline extension/src/popup/components/sendPayment/SendTo/index.tsx:110]: var rename to reflect we can send to either a public key or a C address

**Review by aristidesstaffieri — APPROVED** [review]

**Pull request merged**